### PR TITLE
Add repeatable challenge-related challenge properties

### PR DESCRIPTION
**Pull request by denkv**
As discussed in #7.

### CODE REVIEW

File: ontology.ttl [hunk 1]
```diff
@@ -160,7 +160,28 @@ hobbit:organizer
 		rdfs:label "Organizer"^^xsd:string ;
 		rdfs:domain hobbit:Challenge ;
 		rdfs:range xsd:string .
-				
+
+hobbit:registrationCutoffDate
+		rdf:type owl:DatatypeProperty ;
+		rdfs:comment "Date after which no more changes to the systems participating can be made. No more experiments will run after this date."^^xsd:string ;
```
Comment (MichaelRoeder) [inline ontology.ttl]: That is not fully correct. I would rephrase it to
`Date after which no more systems can be registered for a challenge.`
Experiments will run after this date (i.e., the last "wave" of experiments) and as long as the experiment of a certain system hasn't been executed, its images still can be updated.

File: ontology.ttl [hunk 2]
```diff
@@ -160,7 +160,28 @@ hobbit:organizer
...
+		rdfs:label "Registration Cutoff Date"^^xsd:string ;
+		rdfs:domain hobbit:Challenge ;
+		rdfs:range xsd:dateTime .
+
+hobbit:executionPeriod
+		rdf:type owl:DatatypeProperty ;
+		rdfs:comment "Time between consecutive executions."^^xsd:string ;
+		rdfs:label "Period"^^xsd:string ;
+		rdfs:domain hobbit:Challenge ;
+		rdfs:range xsd:duration .
```
Comment (MichaelRoeder) [inline ontology.ttl]: I don't know enough about this data type to examine whether it is a good choice...
Can you please check whether there is a good representation for it in Java? Otherwise we might use an easier datatype, e.g., `xsd:int` and the meaning "time in hours".

Comment (denkv) [reply ontology.ttl]: Looks like there is [org.apache.jena.datatypes.xsd.XSDDuration](https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/datatypes/xsd/XSDDuration.html) which can be converted (using `toString` representation) to [java.time.Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html).

But `java.util.Calendar` we use doesn't seem to work as is with such `Duration` objects without additional conversions. It has an `add` method where we can specify a time unit and the amount, so yes, using integer value (and maybe optional time unit) may be easier to implement (but less universal).

Comment (denkv) [reply ontology.ttl]: The core handling this will look like this:

```java
Duration executionPeriod = RdfHelper.getDurationValue(challengesModel, challenge, HOBBIT.executionPeriod);
dateOfNextExecution.add(Calendar.MILLISECOND, executionPeriod.toMillis());
```

```java
    public static Duration getDurationValue(Model model, Resource subject, Property predicate) {
        if (model == null) {
            return null;
        }
        Literal literal = getLiteral(model, subject, predicate);
        if (literal != null) {
            try {
                Object o = XSDDatatype.XSDduration.parse(literal.getString());
                if (o instanceof XSDDuration) {
                    return Duration.parse(o.toString());
                }
            } catch (Exception e) {
                // nothing to do
                LOGGER.debug("Couldn't parse " + XSDDatatype.XSDduration.getURI() + ". Returning null.", e);
            }
        }
        return null;
    }
```

Need to test it when everything will be in place, so we can merge this later.

**Comment by denkv** [comment]
Should `registrationCutoffDate` be just `xsd:date` without time?

(`dateOfNextExecution` still needs to be `xsd:dateTime`, since `executionPeriod` may be not in days.)

**Comment by MichaelRoeder** [comment]
No, please use `xsd:dateTime` since it contains the timezone as well.

File: ontology.ttl (hunk 2 above)
Comment (denkv) [reply ontology.ttl]: Duration can be represented with `Java.Time.Duration`. While not ideal when used with `Calendar`, looks like it works fine (code already exists in `feature/repeatable-challenge` branches).
